posthog/hogql_queries/ai/traces_query_runner.py

```diff
@@ -0,0 +1,254 @@
+                else:
+                    generation[model_prop] = properties[event_prop]
+
+        return LLMGeneration.model_validate(generation)
+
+    def _map_person(self, person: tuple[UUID, UUID, datetime, str]) -> LLMTracePerson:
+        uuid, distinct_id, created_at, properties = person
+        return LLMTracePerson(
+            uuid=str(uuid),
+            distinct_id=str(distinct_id),
+            created_at=created_at.isoformat(),
+            properties=orjson.loads(properties),
+        )
+
+    def _get_select_fields(self) -> list[ast.Expr]:
+        return [
+            ast.Alias(expr=ast.Field(chain=["properties", "$ai_trace_id"]), alias="id"),
+            ast.Alias(expr=ast.Call(name="min", args=[ast.Field(chain=["timestamp"])]), alias="trace_timestamp"),
+            self._get_person_field(),
+            ast.Alias(
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["properties", "$ai_latency"])]),
+                alias="total_latency",
+            ),
+            ast.Alias(
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["properties", "$ai_input_tokens"])]),
+                alias="input_tokens",
+            ),
+            ast.Alias(
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["properties", "$ai_output_tokens"])]),
+                alias="output_tokens",
+            ),
+            ast.Alias(
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["properties", "$ai_input_cost_usd"])]),
+                alias="input_cost",
+            ),
+            ast.Alias(
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["properties", "$ai_output_cost_usd"])]),
+                alias="output_cost",
+            ),
+            ast.Alias(
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["properties", "$ai_total_cost_usd"])]),
+                alias="total_cost",
+            ),
+            ast.Alias(
+                expr=ast.Call(
+                    name="arraySort",
+                    args=[
+                        ast.Lambda(
+                            args=["x"],
+                            expr=ast.Call(name="tupleElement", args=[ast.Field(chain=["x"]), ast.Constant(value=2)]),
+                        ),
+                        ast.Call(
+                            name="groupArray",
+                            args=[
+                                ast.Tuple(
+                                    exprs=[
+                                        ast.Field(chain=["uuid"]),
+                                        ast.Field(chain=["timestamp"]),
+                                        ast.Field(chain=["properties"]),
+                                    ]
+                                )
+                            ],
+                        ),
+                    ],
+                ),
+                alias="events",
+            ),
```

Comment:
The rest of those columns is pretty readable in direct AST form, but in this one's case I propose `parse_expr` – it's a chonker